File: Strings/Max Chars/max_chars.py


def max_chars(string):
    chars_count_map = {}
    max_char = ''
    max_char_count = 0
    for c in string:
        if chars_count_map.get(c) is None:
            chars_count_map[c] = 1
        chars_count_map[c] += 1

        if(chars_count_map[c] > max_char_count):
            max_char_count = chars_count_map[c]
            max_char = c

    return max_char


print(max_chars("abcdadsfadgadsssssssa") == 's')

File: Strings/Max_Chars/MaxChars.java
package Strings.Max_Chars;

import java.util.Map;
import java.util.HashMap;

public class MaxChars {
    public static void main(String... args) {
        System.out.println(maxChars("abcdadsfadgadsssssssa").equals("s"));
    }

    public static String maxChars(String s) {

        Map<Character, Integer> charsCounter = new HashMap<>();
        String maxChar = "";
        int maxCharCount = 0;

        for (int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);
            if (!charsCounter.containsKey(c)) {
                charsCounter.put(c, 1);
            }

            charsCounter.compute(c, (k, v) -> v + 1);

            if (charsCounter.get(c) > maxCharCount) {
                maxChar = "" + c;
                maxCharCount = charsCounter.get(c);
            }

        }

        return maxChar;

    }

}